### PR TITLE
Javascript javascript3 week3

### PR DESCRIPTION
**Pull request by andrii1**

### CODE REVIEW

File: javascript/javascript3/week2/promise-visual/main.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+function translateAllAtOnce() {
+  Promise.all([
+    moveElement(redBox, redBoxTarget),
+    moveElement(blueBox, blueBoxTarget),
+    moveElement(greenBox, greenBoxTarget),
+  ]).then(() => {
+    console.log("moved all Boxes together");
+  });
+}
```
Comment (27ManishaPatel) [inline javascript/javascript3/week2/promise-visual/main.js:58]: good work!

File: javascript/javascript3/week3/app.js [hunk 2]
```diff
@@ -15,3 +15,53 @@ context.strokeStyle = "#003300";
...
+
+//running every 100ms
+function generateRandom(maxLimit = 100) {
+  let rand = Math.random() * maxLimit;
+  console.log(rand); // say 99.81321410836433
+
+  rand = Math.floor(rand); // 99
+
+  return rand;
+}
```
Comment (27ManishaPatel) [inline javascript/javascript3/week3/app.js:48]: good idea to create function for random numbers and call it for height, width and radius!

File: javascript/javascript3/week3/app.js [hunk 3]
```diff
@@ -65,3 +61,31 @@ function drawCircle() {
...
+      ul.appendChild(liUser);
+      user.items.forEach((repo) => {
+        console.log(repo);
+        const liRepository = document.createElement("li");
+        liRepository.innerHTML = `${repo.name} ${repo.url}`;
+        ul.appendChild(liRepository);
+        github.appendChild(ul);
+      });
+    });
+  });
```
Comment (27ManishaPatel) [inline javascript/javascript3/week3/app.js:91]: Perfect !

File: javascript/javascript3/week3/shopping-cart.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    // Implement functionality here
+    const sum = this.products.reduce((accumulator, object) => {
+      return accumulator + object.price;
+    }, 0);
+    const renderedProducts = document.getElementById("products");
+    const p = document.createElement("p");
+    p.innerHTML = `Total sum: ${sum}`;
+    renderedProducts.appendChild(p);
+  }
+
```
Comment (27ManishaPatel) [inline javascript/javascript3/week3/shopping-cart.js:55]: Here we can simply  return total value as ' return  this.products.reduce((accumulator, object) => { accumulator + object.price}, 0)'
and DOM functionality in the renderProducts function with this.getTotal()

File: javascript/javascript3/week3/shopping-cart.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  renderProducts() {
+    // Implement functionality here
+
```
Comment (27ManishaPatel) [inline javascript/javascript3/week3/shopping-cart.js:58]: you can remove this comment ' // Implement functionality here' in whole file 😊

File: javascript/javascript3/week2/app.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+const resolveAfterTime = (resolveAfter) =>
+  new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, resolveAfter * 1000);
+  });
+resolveAfterTime(3).then(() => {
+  console.log("I am called asynchronously"); // logged out after 8 seconds
+});
```
Comment (shanhaider23) [inline javascript/javascript3/week2/app.js:11]: Good work :) Another way
function resolveAfterTime(resolveAfter) {
return new Promise((resolve, reject) => {
setTimeout(() => {
resolve();
}, resolveAfter * 1000);
});
}
resolveTimeout(3).then(() => {
console.log("I am called asynchronously.");
});

File: javascript/javascript3/week2/promise-visual/main.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    })
+    .then(() => moveElement(blueBox, blueBoxTarget))
+    .then(() => {
+      console.log("moved blue Box");
+    })
+    .then(() => moveElement(greenBox, greenBoxTarget))
+    .then(() => {
+      console.log("moved green Box");
+    });
+
```
Comment (shanhaider23) [inline javascript/javascript3/week2/promise-visual/main.js:34]: Another way 
async function translateOneByOne() {
  try {
    await moveElement(redBox, redBoxTarget);
    red.innerHTML = `Red box moved`;
    await moveElement(blueBox, blueBoxTarget);
    blue.innerHTML = `Blue box moved`;
    await moveElement(greenBox, greenBoxTarget);
    green.innerHTML = `Green box moved`;
  } catch (error) {
    console.log(error.message);
  }
}
translateOneByOne();

**Comment by shanhaider23** [comment]
Impressive work :) write some comments for info

**Review by shanhaider23 — APPROVED** [review]

**Pull request merged**